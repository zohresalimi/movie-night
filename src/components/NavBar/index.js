import React, { useState } from "react";
import {
  Grid,
  Icon,
  Image,
  Menu,
  Sidebar,
  Responsive,
} from "semantic-ui-react";
import NavBarMobile from "../NavBarMobile";
import NavBarDesktop from "../NavBarDesktop";

export default function NavBar(props) {
  const [visible, setVisible] = useState(false);
  const handleToggle = () => setVisible(!visible);
  const handlePusher = () => {
    if (visible) setVisible(false);
  };
  return (
    <Grid columns={1}>
      <Grid.Row only="mobile">
        <Grid.Column>
          <NavBarMobile
            onPusherClick={handlePusher}
            onToggle={handleToggle}
            visible={visible}
          ></NavBarMobile>
        </Grid.Column>
      </Grid.Row>
      <Grid.Row only="computer tablet">
        <Grid.Column>
          <NavBarDesktop />
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
}
