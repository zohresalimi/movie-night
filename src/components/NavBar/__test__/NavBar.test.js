import * as React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import NavBar from "../";

jest.mock("../../../components/NavBarMobile", () => (props) => (
  <div visible={props.visible.toString()}>Navbar Mobile</div>
));

jest.mock("../../../components/NavBarDesktop", () => () => (
  <div>Navbar Desktop</div>
));

describe("NavBar Component Testing", () => {
  test("take snapshot", async () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });

  test("match snapshot if visible is true", async () => {
    const setState = jest.fn();
    jest.spyOn(React, "useState").mockImplementation(() => [true, setState]);
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
