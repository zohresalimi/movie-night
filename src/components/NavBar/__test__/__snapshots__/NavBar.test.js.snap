// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component Testing match snapshot if visible is true 1`] = `
<div>
  <div
    class="ui one column grid"
  >
    <div
      class="mobile only row"
    >
      <div
        class="column"
      >
        <div
          visible="true"
        >
          Navbar Mobile
        </div>
      </div>
    </div>
    <div
      class="computer only tablet only row"
    >
      <div
        class="column"
      >
        <div>
          Navbar Desktop
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavBar Component Testing take snapshot 1`] = `
<div>
  <div
    class="ui one column grid"
  >
    <div
      class="mobile only row"
    >
      <div
        class="column"
      >
        <div
          visible="false"
        >
          Navbar Mobile
        </div>
      </div>
    </div>
    <div
      class="computer only tablet only row"
    >
      <div
        class="column"
      >
        <div>
          Navbar Desktop
        </div>
      </div>
    </div>
  </div>
</div>
`;
